### PR TITLE
[Observation list] Remove form tab header

### PR DESCRIPTION
**Pull request by gino-m**

### CODE REVIEW

File: gnd/build.gradle [hunk 1]
```diff
@@ -205,7 +205,3 @@ apply plugin: 'com.google.gms.google-services'
 task checkCode(type: GradleBuild) {
     dependsOn 'checkstyle', 'lintDebug', 'pmd', 'spotbugs'
 }
-
-afterEvaluate {
```
Comment (scolsen) [inline gnd/build.gradle]: This seems unrelated to the overall PR--what's the impetus for this change?

Looks like this lets us assemble a debug build w/o enforcing style etc. checks?

LGTM, just calling it out in case it slipped into this PR accidentally.

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java [hunk 2]
```diff
@@ -148,31 +147,28 @@ public void onBottomSheetHidden() {
...
-      Log.e(TAG, "Missing form");
-      return;
-    }
     Feature feature = state.getFeature();
     if (feature == null) {
       Log.e(TAG, "Missing feature");
       return;
     }
+    ImmutableList<Form> forms = feature.getLayer().getForms();
+    if (forms.isEmpty()) {
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java]: Add TODO to show a message to the user?

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/FeatureSheetFragment.java [hunk 3]
```diff
@@ -76,23 +69,13 @@ public View onCreateView(
   @Override
   public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
     super.onViewCreated(view, savedInstanceState);
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/FeatureSheetFragment.java]: This method is redundant now -- delete?

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java [hunk 4]
```diff
@@ -59,14 +59,22 @@ public ObservationListViewModel(ObservationRepository observationRepository) {
   }
 
   /** Loads a list of observations associated with a given feature. */
-  public void loadObservationList(Feature feature, Form form) {
+  public void loadObservationList(Feature feature) {
+    ImmutableList<Form> forms = feature.getLayer().getForms();
     loadObservations(
-        feature.getProject(), feature.getLayer().getId(), form.getId(), feature.getId());
+        feature.getProject(),
+        forms.isEmpty() ? Optional.empty() : Optional.of(forms.get(0).getId()),
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java]: Are we eventually going to support multiple forms again, or will `forms` eventually become `form`?

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java [hunk 5]
```diff
@@ -76,22 +84,16 @@ public void loadObservationList(Feature feature, Form form) {
...
   }
 
-  private void loadObservations(Project project, String layerId, String formId, String featureId) {
-    Optional<Form> form = project.getLayer(layerId).flatMap(pt -> pt.getForm(formId));
-    if (!form.isPresent()) {
-      // TODO: Show error.
-      return;
-    }
-    // TODO: Use project id instead of object.
+  private void loadObservations(Project project, Optional<String> formId, String featureId) {
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java]: nit: Parallelize parameter positions? 

loadObservations:: project, formId, featureId
ObservationListRequest: project, featureId, formId

Swap feature and form in one or the other?

File: gnd/build.gradle (hunk 1 above)
Comment (gino-m) [reply gnd/build.gradle]: Actually it did! I removed it because it was slowly down development significantly, but we can discuss this separately. Added it back.

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/FeatureSheetFragment.java (hunk 3 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/FeatureSheetFragment.java]: Fixed!

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java (hunk 5 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java]: Good catch. Done!

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java (hunk 2 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java]: Added TODO to hide Add Observation button instead, since it shouldn't be shown if form is not defined (i.e., this case should never happen).

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java (hunk 4 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java]: The idea is that we'll keep it in the data store schema, but not in the UIs. This way custom or future clients can use multiple forms without breaking the database.

**Comment by gino-m** [comment]
Thanks for the review! RFAL.

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java (hunk 4 above)
Comment (scolsen) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java]: Gotcha, maybe this is worthy of a helper method to make sure we always call `.get(0)` for now to fetch only the first form and don't accidentally call `.get(0±n)`? (Not that I anticipate us needing to call this in many other places in the app atm).

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/ObservationListViewModel.java]: I'll go you one better - pushing the cardinality change to the persistence layers, so that the application model doesn't reflect this idiosyncrasy in the data models. Please see latest commit 56cfa3df66cd7312961db3633c0fd06e919750c1.

**Comment by gino-m** [comment]
@scolsen should be ready for approval!

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/LayerConverter.java [hunk 6]
```diff
@@ -33,10 +35,12 @@ static Layer toLayer(String id, LayerNestedObject obj) {
     if (obj.getDefaultStyle() != null) {
       layer.setDefaultStyle(StyleConverter.toStyle(obj.getDefaultStyle()));
     }
-    if (obj.getForms() != null) {
-      Maps.forEach(
-          obj.getForms(),
-          (formId, formDoc) -> layer.addForm(FormConverter.toForm(formId, formDoc)));
+    if (obj.getForms() != null && !obj.getForms().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/LayerConverter.java]: That's exactly right.. It would if we controlled how the field was populated, but since we don't know if the Firebase API will set a null or empty we need to check both.
